### PR TITLE
feat:merge the json placeholder users and posts  api

### PR DESCRIPTION
**Pull request by motuncoded**
This shows how I merged the json placeholder api for users and posts together.

### CODE REVIEW

File: pages/Posts.tsx [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
+import {
+  FaRegComment,
+  FaRegHeart,
+  FaHeart,
+  FaRegShareSquare,
+} from "react-icons/fa";
 
+//api
 const BASE_URL = "https://jsonplaceholder.typicode.com/posts?_page=";
+const PRY_URL = "https://jsonplaceholder.typicode.com/users";
```
Comment (Asylcreek) [inline pages/Posts.tsx]: What does `PRY` mean?

File: pages/Posts.tsx [hunk 2]
```diff
@@ -1,21 +1,38 @@
...
   type Error = {
     message: string;
   };
   type Page = number;
 
+  type User = {
+    id: number;
+    name: string;
+    username: string;
+  };
```
Comment (Asylcreek) [inline pages/Posts.tsx]: Move these outside of the React function

File: pages/Posts.tsx [hunk 3]
```diff
@@ -77,19 +102,59 @@ export default function Posts() {
       </div>
     );
   }
+  const handleLikeClick = (post: Post) => {
+    setPosts(
+      posts.map((p) =>
```
Comment (Asylcreek) [inline pages/Posts.tsx]:
```suggestion
    setPosts((prevPosts)=>
      prevPosts.map((p) =>
```

File: pages/Posts.tsx [hunk 4]
```diff
@@ -77,19 +102,59 @@ export default function Posts() {
...
+        p.id === post.id
+          ? { ...p, likes: (p.likes || 1) + 0, liked: !p.liked }
```
Comment (Asylcreek) [inline pages/Posts.tsx]: Can you please explain what is happening on this line?

File: pages/Posts.tsx [hunk 5]
```diff
@@ -77,19 +102,59 @@ export default function Posts() {
...
+      ),
+    );
+  };
 
   return (
     <div ref={scrollContainerRef} className="postContainer">
-      <h1>Post Feed</h1>
+      <h2 className="text-[32px]">Post Feed</h2>
       {posts.map((post: Post) => {
+        const user = users.find((user) => user.id === post.id);
```
Comment (Asylcreek) [inline pages/Posts.tsx]: Can you please explain what is happening on this line?

File: pages/Posts.tsx [hunk 6]
```diff
@@ -77,19 +102,59 @@ export default function Posts() {
...
         return (
           <div key={post.id} className="postWrapper">
+            <div className=" flex justify-between items-center pb-2">
+              <h2 className="font-bold flex flex-col">
+                {user?.name}
+                <span className="text-gray font-thin	"> @{user?.username}</span>
+              </h2>
+              <HiDotsHorizontal />
+            </div>
             <h2 className="postTitle">{post.title}</h2>
```
Comment (Asylcreek) [inline pages/Posts.tsx]: I don't think it's wise to have 2 `h2`'s in the same `div`. Is there a reason you did it?

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: pages/Posts.tsx (hunk 1 above)
Comment (motuncoded) [reply pages/Posts.tsx]: just a random name

File: pages/Posts.tsx (hunk 4 above)
Comment (motuncoded) [reply pages/Posts.tsx]: I want the like button to change when clicked and to change back when clicked again like  a toogle

File: pages/Posts.tsx (hunk 5 above)
Comment (motuncoded) [reply pages/Posts.tsx]: to make sure the posts id and the user id are the same to merge the api

Comment (Asylcreek) [reply pages/Posts.tsx]: Is `post.id` not the id of the post? Why would it ever be equal to the id of the user?

Comment (motuncoded) [reply pages/Posts.tsx]: Post.id is for the id of the post, I want it to be according to the user id

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**